src/handlers/feature_handlers.cpp

```diff
@@ -4493,32 +4490,71 @@ void dgBangerInstanceHandler::DrawShadow() {
     }
 }
 
+static ConfigValue<int> cfgLightClippingFix("LightClippingFix", 0);
+
 void dgBangerInstanceHandler::DrawGlow()
 {
+    auto inst = reinterpret_cast<dgBangerInstance*>(this);
+
     //first time texture load
     if (!glowLoaded) {
         redGlowTexture = gfxGetTexture("s_red_glow", true);
         glowLoaded = true;
     }
 
     //prepare glow texture
-    dgBangerData* data = hook::Thunk<0x441AB0>::Call<dgBangerData *>(this);
-    gfxTexture* lastTexture = (gfxTexture*)glowTexture;
+    dgBangerData* data = inst->GetData();
+    gfxTexture* lastTexture = (gfxTexture*)dgBangerInstance::DefaultGlowTexture;
     bool swappedTexture = false;
 
     if (!strcmp(data->GetName(), "sp_light_red_f") && lastTexture != NULL) {
         swappedTexture = true;
-        glowTexture = redGlowTexture;
+        dgBangerInstance::DefaultGlowTexture = redGlowTexture;
     }
 
     //draw glows
     ltLight::DrawGlowBegin();
-    hook::Thunk<0x441840>::Call<void>(this); // call original
+    if ((inst->GetFlags() & 1) != 0)
+    {
+        Matrix34 matrix;
+        Matrix34 bangerMatrix = inst->GetMatrix(&matrix);
+        Vector3 camPosition = *(Vector3*)&gfxRenderState::sm_Camera->m30;
+
+        if (dgBangerInstance::DefaultGlowTexture != nullptr && data != nullptr)
+        {
+            for (int i = 0; i < data->getNumGlows(); i++)
+            {
+                Vector3 glowOffset = data->getGlowOffset(i);
+
+                float lX = glowOffset.X * bangerMatrix.m00 + glowOffset.Y * bangerMatrix.m10 + glowOffset.Z * bangerMatrix.m20 + bangerMatrix.m30;
+                float lY = glowOffset.X * bangerMatrix.m01 + glowOffset.Y * bangerMatrix.m11 + glowOffset.Z * bangerMatrix.m21 + bangerMatrix.m31;
+                float lZ = glowOffset.X * bangerMatrix.m02 + glowOffset.Y * bangerMatrix.m12 + glowOffset.Z * bangerMatrix.m22 + bangerMatrix.m32;
```

Comment:
You need to learn to spot common vector/matrix operations which have been inlined during compilation. I.E this is a dot product, likely from Vector3::Dot